packages/cliqz-home-react/src/SpeedDialsRow.js

```diff
@@ -0,0 +1,160 @@
+import React, { Component } from 'react';
+import SpeedDial from './SpeedDial';
+import AddSpeedDial from './AddSpeedDial';
+import cliqz from './Cliqz';
+
+class SpeedDialsRow extends Component {
+  constructor(props) {
+    super(props);
+
+    // this.closeUndo = this.closeUndo.bind(this);
+    this.undoRemoval = this.undoRemoval.bind(this);
+    this.resetAll = this.resetAll.bind(this);
+  }
+
+  componentWillMount() {
+    this.state = {
+      isCustom: this.props.type === 'custom',
+      showAddButton: () => {
+        return (this.state.isCustom && this.state.displayAddBtn());
+      },
+      displayAddBtn: () => {
+        return this.props.dials.length < 5;
+      },
+      removedItems: []
+    }
+  }
+
+  removeSpeedDial(dial, index) {
+    this.props.removeSpeedDial(dial, index);
+    dial.removedAt = index;
+    this.state.removedItems.push(dial);
+    this.setState({
+      removedItems: this.state.removedItems
+    });
+
+    cliqz.core.sendTelemetry({
+      type: 'home',
+      action: 'click',
+      target_type: 'remove-' + (dial.custom ? 'custom' : 'history'),
+      target_index: index
+    });
+  }
+
+  closeUndo(type) {
+    this.setState({
+      removedItems: []
+    });
+
+    cliqz.core.sendTelemetry({
+      type: 'home',
+      action: 'click',
+      target_type: 'close-' + type + '-undo'
+    });
+  }
+
+  _reAddSpeedDial(dial) {
+    const index = dial.removedAt;
+    this.props.addSpeedDial(dial, index);
+    cliqz.freshtab.addSpeedDial(dial.url, index).then(() => {
+      let newItems = this.state.removedItems.filter((item) => {
+        return item != dial;
+      });
+      this.setState({
+        removedItems: newItems
+      });
+    })
+  }
+
+  undoRemoval() {
+    const speedDial = this.state.removedItems.pop();
+    const type = speedDial.custom ? 'custom' : 'history';
+
+    if(speedDial.custom) {
+      this._reAddSpeedDial.call(this, speedDial);
+    } else {
+      cliqz.freshtab.revertHistorySpeedDial(speedDial.url);
+      this.props.addSpeedDial(speedDial, speedDial.removedAt);
+    }
+
+    cliqz.core.sendTelemetry({
```

Comment:
how about putting all telemetry signals in one file (as we did in `dropdown`) to hide their complexity a little bit?